lib/tools/system-calls.js

```diff
@@ -236,12 +248,19 @@ systemCallMethods.getConnectedDevices = async function getConnectedDevices () {
   const devices = stdout.split('\n')
     .map(_.trim)
     .filter((line) => line && !excludedLines.some((x) => line.includes(x)))
-    .reduce((acc, line) => {
+    .map((line) => {
       // state is "device", afaic
-      const [udid, state] = line.split(/\s+/);
-      acc.push({udid, state});
-      return acc;
-    }, []);
+      const [udid, state, ...description] = line.split(/\s+/);
+      const device = {udid, state};
+      if (opts.long) {
+        for (const entry of description) {
+          // each entry looks like key:value
+          const [key, value] = entry.split(':');
+          device[key] = value;
```

Comment:
Might there be a situation where key name overrides uuid or state?

Reply:
There shouldn't be, I don't believe "serial", "udid", "uuid", or "state" are ever used as keys.

The corresponding ADB code can be found here: https://android.googlesource.com/platform/system/adb/+/refs/heads/master/transport.cpp#906

Keys are product, model, device, as well as devpath related info. `devpath` is harder to track but generally looks like `usb` or `transport`.